statline_bq/utils.py

```diff
@@ -100,17 +107,69 @@ def get_metadata_cbs(urls: dict, odata_version: str) -> dict:
     ValueError
         If odata_version is not "v3" or "v4"
     """
-
     if odata_version == "v3":
-        target_url = "?".join((urls["TableInfos"], "$format=json"))
-        meta = requests.get(target_url).json()["value"][0]
+        catalog_urls = {
+            True: "https://dataderden.cbs.nl/ODataCatalog/Tables?$format=json",
+            False: "https://opendata.cbs.nl/ODataCatalog/Tables?$format=json",
+        }
+        url = catalog_urls[third_party]
+        params = {}
+        params["$filter"] = f"Identifier eq '{id}'"
+        tables = requests.get(url, params).json()["value"]
+        if len(tables) == 1:
+            meta = tables[0]
+        elif len(tables) == 0:
+            # TODO: HANDLE PROPERLY - is ValueError appropriate here?
+            raise ValueError(
+                "Dataset ID not found. Please enter a valid ID, and ensure third_paty is set appropriately"
+            )
+        else:
+            # TODO: HANDLE PROPERLY - is ValueError appropriate here?
+            raise ValueError(
+                "Multiple matches on same dataset ID found. Please enter a valid ID, and ensure third_paty is set appropriately. Also, this is weird and should not happen."
+            )
     elif odata_version == "v4":
-        meta = requests.get(urls["Properties"]).json()
+        if third_party:
+            # TODO: HANDLE PROPERLY - is ValueError appropriate here?
```

Comment:
You could combine all the URLs in a switch dictionairy as follows:

```
switch_catalog =  {
            ('v3', True): "https://dataderden.cbs.nl/ODataCatalog/Tables?$format=json",
            ('v4', False): "https://opendata.cbs.nl/ODataCatalog/Tables?$format=json",
            ('v4', False): f"https://odata4.cbs.nl/CBS/{id}/properties"
        }
```

Using `dict.get` is the Pythonic way to handle non-existing keys:
```
url = switch_catalog.get((odata_version, third_party), None)
if not url:
    raise ValueError("Valid combinations of (odata_version, third_party are ("v3", True), ("v3", False), ("v4", False)")

Reply:
+1